include/seqan3/search/algorithm/configuration/mode.hpp

```diff
@@ -0,0 +1,153 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides the mode configuration to define the search modes "all", "all_best", "best" and "strata".
+ * \author Christopher Pockrandt <christopher.pockrandt AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/algorithm/all.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/search/algorithm/configuration/utility.hpp>
+
+/*!\addtogroup search
+ * \{
+ */
+
+namespace seqan3::detail
+{
+
+struct search_mode_all {};
+
+struct search_mode_all_best {};
+
+struct search_mode_best {};
+
+} // namespace seqan3::detail
+
+namespace seqan3::search_cfg
+{
+
+//!\brief Configuration element to receive all hits within the error bounds.
+inline detail::search_mode_all constexpr all;
+//!\brief Configuration element to receive all hits within the lowest number of errors.
+inline detail::search_mode_all_best constexpr all_best;
+//!\brief Configuration element to receive one best hit (with the lowest number of errors).
+inline detail::search_mode_best constexpr best;
+
+/*!\brief Configuration element to receive all hits with the best number of errors plus the strata value.
+ *        A strong type of underlying type `uint8_t` that represents the number or errors for strata.
+ *        All hits are found with the fewest numbererrors plus 'value'.
+ * \tparam value_t The underlying type
+ * \ingroup search_configuration
+ */
+struct strata : detail::strong_type<uint8_t, strata, detail::strong_type_skill::convert>
+{
+    using detail::strong_type<uint8_t, strata, detail::strong_type_skill::convert>::strong_type;
+};
+
+} // namespace seqan3::search_cfg
+
+namespace seqan3::detail
+{
+/*!\brief Configuration element to receive all possible hits.
+ * \ingroup search_configuration
+ */
+template <typename mode_t>
+struct search_config_mode
+{
+    //!\cond
+    mode_t value;
+    //!\endcond
+};
+
+/*!\brief The seqan3::search_cfg::mode adaptor enabling pipe notation.
+ * \ingroup search_configuration
+ */
+template <template <typename ...> typename search_config_mode_type>
+struct search_config_mode_adaptor : public configuration_fn_base<search_config_mode_adaptor<search_config_mode>>
```

Comment:
Shoudn't it be the `search_config_mode_type ` in the derived_type declaration. 
I mean seems the work, but it is basically not the same template.

Reply:
Good catch! :eyes: